### PR TITLE
Fix some issues with @save and @load

### LINKED ISSUE #253: Error using @save in Main
```
using JLD
a = 1
@save "test.jld"
```
I get the erro:
`ERROR: LoadError: UndefVarError: current_module not defined`

Seems that there is some old 0.6 code here. I will submit a PR.

### PR DESCRIPTION
**Pull request by Thuener**
fix #253

### CODE REVIEW

**Comment by aramirezreyes** [comment]
"@save" and "@load" still fail 15 month later. Could someone with the powers take a look at this and suggest some improvement that makes it ready to merge?

**Comment by JeffBezanson** [comment]
Instead of using `Main`, this should use the hidden argument `__module__` in the macro, which will give the calling context.

File: src/JLD.jl [hunk 1]
```diff
@@ -1150,13 +1150,13 @@ macro save(filename, vars...)
     if isempty(vars)
         # Save all variables in the current module
         writeexprs = Vector{Expr}(undef, 0)
-        m = current_module()
+        m = Main
```
Comment (JeffBezanson) [inline src/JLD.jl:1153]:
```suggestion
        m = __module__
```

File: src/JLD.jl [hunk 2]
```diff
@@ -1185,7 +1185,7 @@ macro load(filename, vars...)
                      unexpected behavior unless the file is specified as a string literal. Future
                      versions of JLD will require that the file is specified as a string literal
                      in this case.""")
-            filename = eval(current_module(), filename)
+            filename = eval(Main, filename)
```
Comment (JeffBezanson) [inline src/JLD.jl:1188]:
```suggestion
            filename = eval(__module__, filename)
```

**Comment by avigliotti** [comment]
I am having the same error

**Pull request merged**